sljit_src/sljitNativeX86_64.c

```diff
@@ -76,7 +76,7 @@ static sljit_u8* generate_far_jump_code(struct sljit_jump *jump, sljit_u8 *code_
 
 static sljit_u8* generate_put_label_code(struct sljit_put_label *put_label, sljit_u8 *code_ptr, sljit_uw max_label)
 {
-	if (max_label > HALFWORD_MAX) {
+	if (max_label > HALFWORD_MAX && put_label->label->addr > HALFWORD_MAX) {
```

Comment:
this is obviously incorrect as the rule should also take into consideration put_label->addr and indeed now throws an assert in Ubuntu 20.04 amd64 with this code and using the regular allocator (in test63) but works fine in macOS.